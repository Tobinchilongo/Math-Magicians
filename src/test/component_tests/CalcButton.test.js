import React from 'react';
import { render, screen } from '@testing-library/react';
import CalcButton from '../../Components/CalculatorBtn';
import calculate from '../../Logic.js/Calculate';

describe('Testing Calculator Button', () => {
  test('Click a Button on Calculator', () => {
    const calcObj = {
      total: null,
      next: null,
      operation: null,
    };

    render(
      <CalcButton text="9" onClick={() => calculate(calcObj, '9')} mathObj={calcObj} />,
    );
    screen.getByText('9');
  });
});
