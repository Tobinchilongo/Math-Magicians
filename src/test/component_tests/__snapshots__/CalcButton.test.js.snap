// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator Button Click a Button on Calculator 1`] = `
<button
  className="calc-buttons"
  onClick={[Function]}
  type="button"
>
  9
</button>
`;
